pennylane/devices/tests/test_templates.py

```diff
@@ -225,6 +225,15 @@ def circuit():
             [math.fidelity_statevector(circuit(), exp_state)], [1.0], atol=tol(dev.shots)
         )
 
+    @pytest.fixture(scope="function", autouse=False)
```

Comment:
Why this is needed at all? Can't we remove the template from the tests as discussed for QubitStateVector [PR](https://github.com/PennyLaneAI/pennylane/pull/6172)?

Reply:
The other PR was only testing a small functionality that could be ignored.  These are more fundamental tests that we must maintain. Mudit explained me the right way to do it so now I'll get on with it

Reply:
Not sure I understand the difference. Then why the same trick used here is not used in other deprecation PRs?

Reply:
Answered in slack, copying here:

[This is another PR ](https://github.com/PennyLaneAI/pennylane/pull/6026/files#)were you can see the "trick"